ext/simplexml/simplexml.c

```diff
@@ -571,7 +571,7 @@ static int sxe_prop_dim_write(zval *object, zval *member, zval *value, zend_bool
 		if (elements) {
 			if (!member || Z_TYPE_P(member) == IS_LONG) {
 				if (node->type == XML_ATTRIBUTE_NODE) {
-					php_error_docref(NULL, E_ERROR, "Cannot create duplicate attribute");
+					zend_throw_error(zend_ce_error, "Cannot create duplicate attribute");
```

Comment:
Added the same check and free below the throw for `value` that is at the end of the function.